docs/reactivity/key-concepts.md

```diff
@@ -0,0 +1,71 @@
+
+An observable can emit different values over time. In Oryx, components are bind to data observed from APIs and stored in the [application state](#application-state). Whenever the application state is updated, a new value is emitted and the component will update its view automatically in an efficient manner.
+
+Oryx makes use of reactive programming through the popular library [RxJS](https://rxjs.dev/). It provides a set of tools and techniques to make it easier to work with asynchronous data streams and event-driven systems.
+
+## Application state
+
+Application state refers to the data that represents the _current_ state of the application. It describes the current state of an application at any given moment and is used to render the user interface and provide the right behavior for user interactions.
+
+Oryx does not come with a global state management layer. Application state is maintained per _domains_. Each domain is concerned with the associated data that is used in the domain.
+
+Some of the application state is maintained below domains. A good example is the internationalization state (aka application context), such as the active language or active currency. The internationalization domain is considered a core domain and can therefor be used inside other domains without introducing cycle dependencies in the system.
+
+## Handling asynchronous data loading
+
+Oryx simplifies working with asynchronous application state and reactivity by handling the complexity under the hood while still allowing customization. It provides vanilla JavaScript packages for services and lower level layers, which can be reused by component developers regardless of their UI framework choice.
+
+Most of the application state is driven by loading data from backend APIs. Oryx provides a number of standardized application layers:
+
+| Layer      | Purpose                                                                                                             |
+| ---------- | ------------------------------------------------------------------------------------------------------------------- |
+| Component  | Renders application state inside UI elements                                                                        |
+| Controller | Resolves application state for the given context so maximize the component reusability                              |
+| Service    | Manage the application state for a certain application domain.                                                      |
+| Adapter    | Loads the data from a specific backend API                                                                          |
```

Comment:
Maybe it's worth to mention `converters` here as long as this was the one of examples in introduction?